### ISSUE TITLE
no way to print original expression if it's put in a temporary variable

### ISSUE THREAD

**Issue by vogler** [goblint/cil#14]
`d_exp` and `dd_exp` print simple expressions like `x == 3` as is, but for expressions that are taken apart and stored into a temporary variable, like e.g. `y >= 6 && y <= 8`:

- `d_exp` prints `tmp___0` (temporary variable the result of the expression is stored in)
- `dd_exp` prints `<boolean expression>`

**Comment by michael-schwarz** [comment]
I don't think `y >= 6 && y <= 8` is an expression in the sense of `Cil.exp`, since it has control flow.
So one would need to introduce a special sort of function that maps this back to its instructions and then turns it back into a C expression.

**Comment by vogler** [comment]
Yes, maybe there's no nice way to do it in CIL since we'd have to wrap `exp` and e.g. add its location. However, would be nice to have some function for it.
